tests/integration/consumer/test_consumer_upgrade_downgrade.py

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright 2025 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limit
+
+import pytest
+from confluent_kafka import ConsumerGroupType, IsolationLevel, KafkaException, TopicPartition
+from confluent_kafka.admin import OffsetSpec
+from tests.common import TestUtils
+
+topic_prefix = "test_consumer_upgrade_downgrade_"
+number_of_partitions = 10
+
+
+def get_group_protocol_type(a, group_id):
+    futureMap = a.describe_consumer_groups([group_id])
+    try:
+        future = futureMap[group_id]
+        g = future.result()
+        return g.type
+    except KafkaException as e:
+        print("Error while describing group id '{}': {}".format(group_id, e))
+    except Exception:
+        raise
+
+
+def list_offsets(a, topic, no_of_partitions):
+    topic_partition_offsets = {}
+    for partition in range(no_of_partitions):
+        topic_partition = TopicPartition(topic, partition)
+        topic_partition_offsets[topic_partition] = OffsetSpec.latest()
+    futmap = a.list_offsets(topic_partition_offsets, isolation_level=IsolationLevel.READ_COMMITTED, request_timeout=30)
+    for partition, fut in futmap.items():
+        try:
+            result = fut.result()
+            print("Topicname : {} Partition_Index : {} Offset : {} Timestamp : {}"
+                  .format(partition.topic, partition.partition, result.offset,
+                          result.timestamp))
+        except KafkaException as e:
+            print("Topicname : {} Partition_Index : {} Error : {}"
+                  .format(partition.topic, partition.partition, e))
+
+
+# def produce_messages(producer, topic, partitions, num_messages):
+#     for i in range(num_messages):
+#         key = "key-{}".format(i)
+#         value = "value-{}".format(i)
+#         partition = i % partitions
+#         producer.produce(topic, key=key, value=value, partition=partition)
+#     producer.flush()
+
+
+def check_consumer(kafka_cluster, consumers, admin_client, topic, expected_protocol):
+    total_msg_read = 0
+    while len(consumers[-1].assignment()) != number_of_partitions // len(consumers):
+        for consumer in consumers:
+            consumer.poll(0.1)
+
+    for consumer in consumers:
+        assert len(consumer.assignment()) == number_of_partitions // len(consumers)
+
+    assert get_group_protocol_type(admin_client, topic) == expected_protocol
+
+    # Produce some messages to the topic
+    kafka_cluster.seed_topic(topic)
+    list_offsets(admin_client, topic, number_of_partitions)
```

Comment:
The `list_offsets` can be removed given we're producing and then consuming all messages, it'll be necessary all of them were produced.